src/solvers/petsc_dm_wrapper.C

```diff
@@ -517,7 +517,7 @@ namespace libMesh
 
         // Build the PetscSection and attach it to the DM
         this->build_section(system, section);
-#if PETSC_VERSION_LESS_THAN(3,9,0)
+#if PETSC_VERSION_LESS_THAN(3,12,0)
```

Comment:
This API appeared in 3.10 as I mentioned on your original commit. Setting this Macro for 3.12 should work though.